src/dynimage.rs

```diff
@@ -747,32 +752,54 @@ impl GenericImage for DynamicImage {
     type InnerImage = DynamicImage;
 
     fn put_pixel(&mut self, x: u32, y: u32, pixel: color::Rgba<u8>) {
+        let color::Rgba([r, g, b, a]) = pixel;
+        let (r16, g16, b16, a16) = (r as u16 * 257, b as u16 * 257, g as u16 * 257, a as u16 * 257);
```

Comment:
Could this be `let color::Rgba::<u16>([r16, g16, b16, a16]) = pixel.into_color();` instead so as to not duplicate color conversion logic?

Reply:
Now that I think about it, all of these could be replaced with `into_color`, e.g.,

```rust
 DynamicImage::ImageBgra8(ref mut p) => p.put_pixel(x, y, pixel.into_color()),
```
For the 16bpc types you just need an extra,

```rust
let pixel16: color::Rgba<u16> = pixel.into_color();

...

DynamicImage::ImageRgba16(ref mut p) => p.put_pixel(x, y, pixel.into_color()),
```

This is consistent with the two-level API described above in the other comment, with `IntoColor` being a high level trait expressing convertibility across color types that lets `DynamicImage` not have to be responsible for how these conversions are done. This means the `FromColor`/`IntoColor` implementation will be responsible for handling color *space* conversions once we represent them, as they should. The downside for now is still needing to special case the 16bpc types, but I expect we'll clear that up later with the `Pixel` types themselves. 

To note this means that `put_pixel` could really be

```rust
fn<P: Pixel> put_pixel(&mut self, x: u32, y: u32, pixel: P)
    where P: IntoColor<color::Rgba<u8>> + IntoColor<color::Rgba<u16>> {
    let pixel8: color::Rgba<u8> = pixel.into_color();
    let pixel16: color::Rgba<u16> = pixel.into_color();
    ...
}
```